tests/test_keys.py

```diff
@@ -197,6 +197,29 @@ def test_format_metadata_to_key(self):
 
 
 
+  def test_format_metadata_to_key_non_default_hash_algorithms(self):
+    # Reconfiguring rsakey_dict to conform to KEY_SCHEMA
+    # i.e. {keytype: 'rsa', keyval: {public: pub_key, private: priv_key}}
+    keyid = self.rsakey_dict['keyid']
+    del self.rsakey_dict['keyid']
+
+    rsakey_dict_from_meta, junk = KEYS.format_metadata_to_key(self.rsakey_dict,
+      keyid_hash_algorithms=['sha384'])
+
+    # Check if the format of the object returned by this function corresponds
+    # to RSAKEY_SCHEMA format.
+    self.assertEqual(None,
+        securesystemslib.formats.RSAKEY_SCHEMA.check_match(rsakey_dict_from_meta),
+        FORMAT_ERROR_MSG)
+
+    self.assertEqual(None,
+        securesystemslib.formats.KEY_SCHEMA.check_match(rsakey_dict_from_meta),
+        FORMAT_ERROR_MSG)
+
+    self.rsakey_dict['keyid'] = keyid
```

Comment:
As above, we can avoid this if we make a local copy of the dict to use during this test.

Reply:
Addressed your comment by making changes and amending them to my last commit.